cartographer/mapping_3d/sparse_pose_graph.cc

```diff
@@ -106,16 +107,16 @@ void SparsePoseGraph::AddScan(
       transform::Rigid3d::Identity()});
   trajectory_nodes_.push_back(
       mapping::TrajectoryNode{&constant_node_data_.back(), optimized_pose});
-  trajectory_connectivity_.Add(trajectory);
+  trajectory_connectivity_.Add(trajectory_id);
 
   if (submap_indices_.count(insertion_submaps.back()) == 0) {
     submap_indices_.emplace(insertion_submaps.back(),
                             static_cast<int>(submap_indices_.size()));
     submap_states_.emplace_back();
     submap_states_.back().submap = insertion_submaps.back();
-    submap_states_.back().trajectory = trajectory;
+    submap_states_.back().trajectory_id = trajectory_id;
```

Comment:
Similar to 2D: Remove the member from the struct. Similar for other 2D related comments.